### DISCUSSION TITLE
v3 use table without eloquent

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#7160] [Help]
Is it possible to use the table component without Eloquent? I can't seem to find it on the [page](https://beta.filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component).
I would like to not pass a query but have custom data retrieval logic in the backend.

**Comment by divdax** [comment]
You could use the 🍣 [Sushi](https://github.com/calebporzio/sushi) package. For v3 there is already an issue: #7105

**Reply by pboivin** [reply]
There's a complete example here : https://filamentphp.com/blog/how-to-consume-an-external-api-with-filament-tables

**Reply by Wulfheart** [reply]
@pboivin Thanks for the link. As far as I can see this does not support pagination. Does it?

**Reply by pboivin** [reply]
Yes it does, it'll work like a regular table since Sushi emulates a full Eloquent model.

**Reply by nikspyratos** [reply]
Note for v3: The site was revamped, so the link above is now https://v2.filamentphp.com/blog/how-to-consume-an-external-api-with-filament-tables

**Reply by danharrin** [reply]
It's actually on our new site too https://filamentphp.com/community/how-to-consume-an-external-api-with-filament-tables

**Reply by nikspyratos** [reply]
2024: New link again :) https://filamentphp.com/content/leandrocfe-how-to-consume-an-external-api-with-filament-tables
